### DISCUSSION TITLE
CSS custom properties don't work in object-based style notation

### DISCUSSION THREAD

**Discussion by pohlt** [alpinejs/alpine#1807] [7. Bugs (archived)] [answered]
Please have a look at this [example](https://codepen.io/pohlt/pen/RwVgbYX):
The CSS custom property does not work in the object-based notation, but does work when using a template literal.

Is this the expected behavior or am I doing something wrong?

Cheers

**Comment by KevinBatdorf** [comment, marked as answer]
Line 16 should probably be `el.style.setProperty(key, value)`

https://github.com/alpinejs/alpine/blob/main/packages/alpinejs/src/utils/styles.js#L10-L28

**Reply by pohlt** [reply]
Thanks for the hint! 👍🏼 
I tried `el.style.setProperty(key, value2);` (mind the `value2`) and it worked.

Do you want to open a PR?

**Reply by KevinBatdorf** [reply]
Yeah but I'd also add some tests. I'm not sure what coverage there is already but you may want to also make sure the other cases are tested too (like line 20)

**Reply by pohlt** [reply]
Ah, thanks a lot! I'm a noob with JS and wouldn't know how to implement the tests correctly. Stay safe!

**Reply by KevinBatdorf** [reply]
Ah, I thought you were asking me whether you should :D 

Maybe someone else will after seeing the opportunity. I can this week if I have time otherwise. 

But if you want to, you would add tests here: https://github.com/alpinejs/alpine/blob/main/tests/cypress/integration/directives/x-bind.spec.js

And if you do and want me to look just tag me

**Reply by pohlt** [reply]
Just posted a PR: https://github.com/alpinejs/alpine/pull/1854
